timely_beliefs/beliefs/utils.py

```diff
@@ -694,21 +689,29 @@ def interpret_special_read_cases(
     elif len(df.columns) == 3:
         # datetimes in 1st and 2nd column, and value in 3rd column
         df.columns = ["event_start", "belief_time", "event_value"]
-        df["event_start"] = pd.to_datetime(df["event_start"])
-        if df["event_start"].dt.tz is None:
-            df["event_start"] = df["event_start"].dt.tz_localize(
-                timezone, ambiguous="infer"
-            )
-        df["event_start"] = df["event_start"].dt.tz_convert(sensor.timezone)
-        df["belief_time"] = pd.to_datetime(df["belief_time"])
-        if df["belief_time"].dt.tz is None:
-            df["belief_time"] = df["belief_time"].dt.tz_localize(
-                timezone, ambiguous="infer"
-            )
-        df["belief_time"] = df["belief_time"].dt.tz_convert(sensor.timezone)
+        df["event_start"] = to_sensor_timezone(df["event_start"], sensor, timezone)
+        df["belief_time"] = to_sensor_timezone(df["belief_time"], sensor, timezone)
     return df
 
 
+def to_sensor_timezone(
+    s: pd.Series, sensor: "classes.Sensor", timezone: Optional[str]
+) -> pd.Series:
+    """Localize or convert the timezone of the series to the timezone of the sensor."""
+    # Convert to datetime (works for timezone naive datetimes, and timezone aware datetime with a shared offset)
+    s = pd.to_datetime(s)
+    if s.dtype == "object":
```

Comment:
It is a proxy. This condition is telling us that line 702 didn't do its job, otherwise the dtype would have been `datetime64[ns]` (when converting naive datetimes) or `datetime64[ns, pytz.FixedOffset(60)]` (or some other offset, when converting aware datetimes with a shared offset). Mixed dtypes turn up as `object ` in Pandas ([ref](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dtypes.html)).

Reply:
Can we add a short comment about this expectation?